docker/Dockerfile.ng

```diff
@@ -0,0 +1,213 @@
+
+# Old Dependencies for reference
+#apt-get update
+#$minimal_apt_get_install build-essential checkinstall git-core \
+#  zlib1g-dev libyaml-dev libssl-dev libgdbm-dev libreadline-dev \
+#  libncurses5-dev libffi-dev libxml2-dev libxslt-dev libcurl4-openssl-dev libicu-dev \
+#  graphviz libgraphviz-dev \
+#  libmysqlclient-dev libpq-dev libsqlite3-dev \
+#  ruby2.3 ruby2.3-dev
+
+# TODO: Put together a little script that will easily parse out all the .env.example lines to copy into here
+
+# See .env.example for descriptions/usage
+ENV APP_SECRET_TOKEN=REPLACE_ME_NOW! \
+    #ON_HEROKU=true \
+    PORT=3000 \
+    DOMAIN=localhost:3000 \
+    ASSET_HOST= \
+    DATABASE_ADAPTER=mysql2 \
+    DATABASE_ENCODING=utf8m4 \
+    DATABASE_RECONNECT=true \
+    DATABASE_NAME=huginn \
+    DATABASE_POOL=20 \
+    DATABASE_USERNAME=huginn \
+    DATABASE_PASSWORD=huginn \
+    DATABASE_HOST=db \
+    DATABASE_PORT=3306 \
+    DATABASE_SOCKET= \
+    #RAILS_ENV=production \
+    FORCE_SSL=false \
+    INVITATION_CODE=try-huginn \
+    SKIP_INVITATION_CODE=false \
+    REQUIRE_CONFIRMED_EMAIL=false \
+    ALLOW_UNCONFIRMED_ACCESS_FOR=2.days \
+    CONFIRM_WITHIN=3.days \
+    MIN_PASSWORD_LENGTH=8 \
+    RESET_PASSWORD_WITHIN=6.hours \
+    LOCK_STRATEGY=failed_attempts \
+    MAX_FAILED_LOGIN_ATTEMPTS=10 \
+    UNLOCK_STRATEGY=both \
+    UNLOCK_AFTER=1.hour \
+    REMEMBER_FOR=4.weeks \
+    IMPORT_DEFAULT_SCENARIO_FOR_ALL_USERS=true \
+    DEFAULT_SCENARIO_FILE= \
+    SMTP_DOMAIN= \
+    SMTP_USER_NAME= \
+    SMTP_PASSWORD= \
+    SMTP_SERVER=smtp.gmail.com \
+    SMTP_PORT=587 \
+    SMTP_AUTHENTICATION=plain \
+    SMTP_ENABLE_STARTTLS_AUTO=true \
+    SEND_EMAIL_IN_DEVELOPMENT=false \
+    EMAIL_FROM_ADDRESS=from_address@gmail.com \
+    AGENT_LOG_LENGTH=200 \
+    TWITTER_OAUTH_KEY= \
+    TWITTER_OAUTH_SECRET= \
+    THIRTY_SEVEN_SIGNALS_OAUTH_KEY= \
+    THIRTY_SEVEN_SIGNALS_OAUTH_SECRET= \
+    GITHUB_OAUTH_KEY= \
+    GITHUB_OAUTH_SECRET= \
+    TUMBLR_OAUTH_KEY= \
+    TUMBLR_OAUTH_SECRET= \
+    DROPBOX_OAUTH_KEY= \
+    DROPBOX_OAUTH_SECRET= \
+    WUNDERLIST_OAUTH_KEY= \
+    WUNDERLIST_OAUTH_SECRET= \
+    EVERNOTE_OAUTH_KEY= \
+    EVERNOTE_OAUTH_SECRET= \
+    USE_EVERNOTE_SANDBOX=false \
+    AWS_ACCESS_KEY_ID= \
+    AWS_ACCESS_KEY= \
+    AWS_SANDBOX=false \
+    ADDITIONAL_GEMS= \
+    FARADAY_HTTP_BACKEND=typhoeus \
+    DEFAULT_HTTP_USER_AGENT="Huginn - https://github.com/cantino/huginn" \
+    ALLOW_JSONPATH_EVAL=false \
+    ENABLE_INSECURE_AGENTS=false \
+    ENABLE_SECOND_PRECISION_SCHEDULE=false \
+    SCHEDULER_FREQUENCY=0.3 \
+    EVENT_EXPIRATION_CHECK=6h \
+    USE_GRAPHVIZ_DOT= \
+    DIAGRAM_DEFAULT_LAYOUT= \
+    TIMEZONE="Pacific Time (US & Canada)" \
+    FAILED_JOBS_TO_KEEP=100 \
+    DELAYED_JOB_MAX_RUNTIME=2 \
+    DELAYED_JOB_SLEEP_DELAY=10
+
+COPY . "$HUGINN_HOME"
+
+# TODO: Need to tune the .dockerignore, or prevent some things being copied here
+#  eg.
+#    Gemfile.lock has caused issues in testing
```

Comment:
What were those issues? I think the `Gemfile.lock` is essential to ensure the image uses our locked gem version.